cmd/mimir/config-descriptor.json

```diff
@@ -3063,6 +3074,17 @@
           "fieldType": "duration",
           "fieldCategory": "experimental"
         },
+        {
+          "kind": "field",
+          "name": "separate_metrics_label",
```

Comment:
Nit: what if we call this `separate_metrics_group_label`? Given @osg-grafana's feedback below, this would make the setting clearer IMO.

(...or maybe `group_separation_label`? no, I don't like it because it's under `validation` section, so the flag would be `validation.group-separation-label`)

Maybe just `metrics_group_label` so it would be `validation.metrics-group-label`?